addons/im_livechat/static/src/views/discuss_channel_list/discuss_channel_list_view_controller.js

```diff
@@ -1,16 +1,29 @@
 /* @odoo-module */
 
-import { _t } from "@web/core/l10n/translation";
+import { useState } from "@odoo/owl";
+
 import { ListController } from "@web/views/list/list_controller";
+import { useService } from "@web/core/utils/hooks";
+import { _t } from "@web/core/l10n/translation";
 
 export class DiscussChannelListController extends ListController {
     setup() {
         super.setup(...arguments);
+        this.threadService = useService("mail.thread");
+        this.store = useState(useService("mail.store"));
+        this.ui = useState(useService("ui"));
     }
 
     async openRecord(record) {
-        if (!record.data.is_member) {
-            return super.openRecord(record);
+        if (this.ui.isSmall) {
+            let thread = this.store.Thread.get({
+                model: "discuss.channel",
+                id: record.resId,
+            });
+            if (!thread) {
+                thread = await this.threadService.fetchChannel(record.resId);
```

Comment:
`fetchChannel` might return no thread (almost impossible in this flux as the user just click on that thread, but unfortunate race conditions with another user deleting it in the meantime are always possible), this needs to be handled